fri/src/verifier/channel.rs

```diff
@@ -96,24 +96,19 @@ pub trait VerifierChannel<E: FieldElement> {
     ///
     /// # Errors
     /// Returns an error if:
-    /// - Remainder values read from the channel cannot be used to construct a fully-balanced
-    ///   Merkle tree.
-    /// - If the root of the Merkle tree constructed from the remainder values does not match
-    ///   the specified `commitment`.
+    /// - If the sequrntial hash constructed from the remainder values does not match the specified
+    ///   `commitment`.
     fn read_remainder<const N: usize>(
         &mut self,
         commitment: &<<Self as VerifierChannel<E>>::Hasher as Hasher>::Digest,
     ) -> Result<Vec<E>, VerifierError> {
         let remainder = self.take_fri_remainder();
 
-        // build remainder Merkle tree
-        let remainder_values = transpose_slice(&remainder);
-        let hashed_values = hash_values::<Self::Hasher, E, N>(&remainder_values);
-        let remainder_tree = MerkleTree::<Self::Hasher>::new(hashed_values)
-            .map_err(|err| VerifierError::RemainderTreeConstructionFailed(format!("{err}")))?;
+        // sequential hash of the remainder
+        let set_commitment =
+            <<Self as VerifierChannel<E>>::Hasher as ElementHasher>::hash_elements(&remainder);
```

Comment:
Do we still need `RemainderTreeConstructionFailed` error variant somewhere? If not, I'd delete it.

Reply:
Yes, fixed!